src/core/psi_container.jl

```diff
@@ -222,8 +262,146 @@ model_initial_time(psi_container::PSIContainer) = psi_container.initial_time
 #Internal Variables, Constraints and Parameters accessors
 get_variables(psi_container::PSIContainer) = psi_container.variables
 get_constraints(psi_container::PSIContainer) = psi_container.constraints
-get_variable(psi_container::PSIContainer, name::Symbol) = psi_container.variables[name]
-get_constraint(psi_container::PSIContainer, name::Symbol) = psi_container.constraints[name]
 get_parameters(psi_container::PSIContainer, param_reference::UpdateRef) = psi_container.parameters[param_reference]
 get_expression(psi_container::PSIContainer, name::Symbol) = psi_container.expressions[name]
 get_initial_conditions(psi_container::PSIContainer) = psi_container.initial_conditions
+
+function get_variable(
+    psi_container::PSIContainer,
+    var_type::AbstractString,
+    device_type::Type{<:PSY.Device},
+)
+    return get_variable(psi_container, variable_name(var_type, device_type))
+end
+
+function get_variable(psi_container::PSIContainer, var_type::AbstractString)
+    return get_variable(psi_container, variable_name(var_type))
+end
+
+function get_variable(psi_container::PSIContainer, name::Symbol)
+    var = get(psi_container.variables, name, nothing)
+    if isnothing(var)
+        @error "$name is not stored" sort!(get_variable_names(psi_container))
+        throw(IS.InvalidValue("variable $name is not stored"))
+    end
+        
+    return var
+end
+
+function get_variable_names(psi_container::PSIContainer)
+    return collect(keys(psi_container.variables))
+end
+
+function set_variable!(
+    psi_container::PSIContainer,
+    variable_type::AbstractString,
+    device_type::Type{<:PSY.Device},
+    value,
+)
+    set_variable!(psi_container, variable_name(variable_type, device_type), value)
+    return
+end
+
+function set_variable!(
```

Comment:
we should change this name to `assign` or `fill` because in the optimization lingo this can get confused with setting the variable value